set.mm

```diff
@@ -666791,6 +667094,160 @@ D Fn ( I ... ( M - 1 ) ) /\
+    evlsvvvallem2.w $e |- .^ = ( .g ` M ) $.
+    evlsvvvallem2.x $e |- .x. = ( .r ` S ) $.
+    evlsvvvallem2.i $e |- ( ph -> I e. V ) $.
+    evlsvvvallem2.s $e |- ( ph -> S e. CRing ) $.
+    evlsvvvallem2.r $e |- ( ph -> R e. ( SubRing ` S ) ) $.
+    evlsvvvallem2.f $e |- ( ph -> F e. B ) $.
+    evlsvvvallem2.a $e |- ( ph -> A e. ( K ^m I ) ) $.
+    $( Lemma for theorems using ~ evlsvvval .  (Contributed by SN,
+       8-Mar-2025.) $)
+    evlsvvvallem2 $p |- ( ph -> ( b e. D |->
+      ( ( F ` b ) .x. ( M gsum ( v e. I |-> ( ( b ` v ) .^ ( A ` v ) ) ) ) ) )
+                                                        finSupp ( 0g ` S ) ) $=
+      ( cv cfv co cmpt cgsu c0g cvv wcel ccnv cn cima cfn cn0 cmap rabex2 mptex
+      ovex a1i fvexd wfun funmpt eqid cress ovexi mplelsfi csupp cdif wa mplelf
+      cbs ssidd suppssrg csubrg subrg0 syl eqcomd adantr eqtrd oveq1d crngringd
+      wceq eldifi ccrg simpr evlsvvvallem sylan2 ringlzd suppss2 fsuppsssuppgd
+      crg ) AMRERULZMUMZPBNBULZXBUMXDCUMLUNUOUPUNZIUNZUOZJUQUMZURURHUQUMZXGURUS
+      AREXFKULUTVAVBVCUSKVDNVEUNESVDNVEVHVFZVGVIAHUQVJXGVKAREXFVLVIADFJMNURXHTU
+      BXHVMUJJURUSAJHGVNUAVOVIVPAEXFRURMXHVQUNZXIAXBEXKVRUSZVSZXFXIXEIUNXIXMXCX
+      IXEIXMXCXHXIAEJWAUMZURMDXKXBXHADEFJKNXNMTXNVMUBSUJVTAXKWBUJAJUQVJWCAXHXIW
+      LXLAXIXHAGHWDUMUSXIXHWLUIGHJXIUAXIVMZWEWFWGWHWIWJXMOHIXEXIUCUFXOAHXAUSXLA
+      HUHWKWHXLAXBEUSZXEOUSXBEXKWMAXPVSBCXBEHKLNOPQSUCUDUEANQUSXPUGWHAHWNUSXPUH
+      WHACONVEUNUSXPUKWHAXPWOWPWQWRWIEURUSAXJVIWSWT $.
+  $}
+
+  ${
+    $d ph a b i x $.  $d P b $.  $d B b $.  $d D a b i p $.  $d K a b i x $.
+    $d K l $.  $d U b h $.  $d F b x $.  $d I a i x $.  $d I b h $.
+    $d S a b i x $.  $d R b x $.  $d ph l m $.  $d A b i l $.  $d D l m $.
+    $d I l m $.  $d U a l $.  $d M a l $.  $d .x. l $.  $d .^ a l $.  $d F l $.
+    $d S l m $.  $d K l m $.  $d M m $.  $d K m $.  $d b i m p $.  $d S p $.
+    $d ph p $.  $d K p $.  $d .^ m p $.  $d I p $.
+    evlsvvval.q $e |- Q = ( ( I evalSub S ) ` R ) $.
+    evlsvvval.p $e |- P = ( I mPoly U ) $.
+    evlsvvval.b $e |- B = ( Base ` P ) $.
+    evlsvvval.u $e |- U = ( S |`s R ) $.
+    evlsvvval.d $e |- D = { h e. ( NN0 ^m I ) | ( `' h " NN ) e. Fin } $.
+    evlsvvval.k $e |- K = ( Base ` S ) $.
+    evlsvvval.m $e |- M = ( mulGrp ` S ) $.
+    evlsvvval.w $e |- .^ = ( .g ` M ) $.
+    evlsvvval.x $e |- .x. = ( .r ` S ) $.
+    evlsvvval.i $e |- ( ph -> I e. V ) $.
+    evlsvvval.s $e |- ( ph -> S e. CRing ) $.
+    evlsvvval.r $e |- ( ph -> R e. ( SubRing ` S ) ) $.
+    evlsvvval.f $e |- ( ph -> F e. B ) $.
+    evlsvvval.a $e |- ( ph -> A e. ( K ^m I ) ) $.
+    $( Give a formula for the evaluation of a polynomial given assignments from
+       variables to values.  (Contributed by SN, 5-Mar-2025.) $)
+    evlsvvval $p |- ( ph -> ( ( Q ` F ) ` A ) = ( S gsum ( b e. D |->
+      ( ( F ` b ) .x.
+                ( M gsum ( i e. I |-> ( ( b ` i ) .^ ( A ` i ) ) ) ) ) ) ) ) $=
```

Comment:
This looks similar to [`evls1fpws`](https://us.metamath.org/mpeuni/evls1fpws.html) which was also a rather long proof. I can imagine how much harder this is with multiple variables! Congrats!